specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/dedicatedhsm.json

```diff
@@ -0,0 +1,762 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Dedicated HSM Resource Provider",
+    "description": "The Azure management API provides a RESTful set of web services that interact with Azure Dedicated HSM RP.",
+    "version": "2024-06-30"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HardwareSecurityModules/dedicatedHSMs/{name}": {
+      "put": {
+        "tags": [
+          "DedicatedHsms"
+        ],
+        "operationId": "DedicatedHsm_CreateOrUpdate",
+        "x-ms-long-running-operation": true,
+        "description": "Create or Update a dedicated HSM in the specified subscription.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DedicatedHsmName"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsm"
+            },
+            "description": "Parameters to create or update the dedicated hsm"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Created dedicated HSM",
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsm"
+            }
+          },
+          "200": {
+            "description": "Created dedicated HSM",
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsm"
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/DedicatedHsmError"
```

Comment:
>#/definitions/DedicatedHsmError [](http://example.com/codeflow?start=23&length=31)

Consider using v5 errorResponse definition for this.

Reply:
Cannot use common definition. The error schema does for this resource does not match the ErrorResponse

Reply:
ErrorResponse of DedicatedHSm.json does not match any of they common ErrorResponses.